// @jeppesen-foreflight/dp-dls-global-angular/fesm2022/jeppesen-foreflight-dp-dls-global-angular-navigation-drawer.mjs
import * as i0 from '@angular/core';
import { EventEmitter, ElementRef, Output, Input, ViewChild, ViewEncapsulation, Component } from '@angular/core';
import * as i5 from '@angular/material/icon';
import { MatIconModule } from '@angular/material/icon';
import { IconRegistryModule } from '@jeppesen-foreflight/dp-dls-global-angular/icon-registry';
import { MatToolbarModule } from '@angular/material/toolbar';
import * as i3 from '@angular/common';
import { CommonModule } from '@angular/common';
import { MatButtonModule } from '@angular/material/button';
import * as i4 from '@angular/material/sidenav';
import { MatSidenavModule } from '@angular/material/sidenav';
import * as i1 from '@angular/router';
import { RouterOutlet, RouterModule } from '@angular/router';
import * as i6 from '@angular/material/list';
import { MatListModule, MatNavList } from '@angular/material/list';
import { MatDividerModule } from '@angular/material/divider';
import { MatMenuModule } from '@angular/material/menu';
import { coerceBooleanProperty } from '@angular/cdk/coercion';
import * as i2 from '@angular/cdk/layout';

const MockSidebarMenuData = [
    {
        sectionHeader: "Section Header Label 1",
        sectionIcon: "apps",
        sectionIconAriaText: "Example icon-button with a bookmark icon",
        sectionMenuItems: [
            {
                label: 'Level 1 destination',
                icon: 'book',
                count: '99+',
                toggled: false,
                children: [
                    { label: 'Level 2 destination', icon: 'chat', count: '99+', toggled: false },
                    { label: 'Level 2 destination', icon: 'download', count: '99+', toggled: false },
                ],
                expanded: false,
            },
            {
                label: 'Level 1 destination',
                icon: 'palette',
                route: '/level1',
                count: '99+',
                expanded: false,
                toggled: false
            },
            {
                label: 'Level 1 destination',
                icon: 'add-circle',
                route: '/level1',
                count: '99+',
                expanded: false,
                toggled: false
            },
        ]
    },
    {
        sectionHeader: "Section Header Label 2",
        sectionIcon: "code",
        sectionIconAriaText: "Example icon-button with a bookmark icon",
        sectionMenuItems: [
            {
                label: 'Level 1 destination',
                icon: 'tune',
                route: '/level1',
                count: '99+',
                expanded: false,
                toggled: false
            },
            {
                label: 'Level 1 destination',
                icon: 'support',
                route: '/level1',
                count: '99+',
                expanded: false,
                toggled: false
            },
        ]
    }
];

/**
 * An enum denoting the different possible variants for the navigation drawer
 * and the side navigation mode.
 *
 * These can keep us from having a number of "magic strings" sprinkled in the
 * code below and repeated for various evaluations.
 *
 * While this enum is available for comparisons, the component input
 * `navDrawerPresentation` uses the `AtmosphereNavDrawerVariant` type, which is
 * a union of strings.
 *
 * Acceptable variants are `default`, `standalone`, and `modal`.
 *
 */
var NavDrawerPresentationEnum;
(function (NavDrawerPresentationEnum) {
    NavDrawerPresentationEnum["DEFAULT"] = "default";
    NavDrawerPresentationEnum["STANDALONE"] = "standalone";
    NavDrawerPresentationEnum["MODAL"] = "modal";
})(NavDrawerPresentationEnum || (NavDrawerPresentationEnum = {}));
// looks like this is whether it's collapsible or not...
// could have the default be collapsible, and the `fixed` flag passed if not
var SideNavMode;
(function (SideNavMode) {
    SideNavMode["SIDE"] = "side";
    SideNavMode["OVER"] = "over";
    SideNavMode["PUSH"] = "push";
})(SideNavMode || (SideNavMode = {}));
/**
 * The Navigation Drawer component represents a left-hand navigation drawer that
 * can conidtionally collapse/expand on hover, be presented as a modal, or be
 * locked open. It is available branded and unbranded, to account for the
 * respective absence or presence of a top bar.
 *
 * The expected behavior for non-modal variants is as follows: Upon page load,
 * the drawer shall appear in its collapsed state. Upon hovering over a manu
 * item, the drawer will expand, and it will collapse when losing focus. If
 * the user drags the drawer open or selects a menu item, then we treat this as
 * an action with intent, and the drawer does NOT collapse on blur.
 *
 * There are a few key properties to be aware of:
 *
 * `navDrawerPresentation` will determine whether or not the navigation drawer
 * renders as the default (unbranded, for use with a top bar) version, the
 * standalone (branded) version, or the modal version. Acceptable values for
 * this input are, intuitively, `default`, `standalone`, or `modal`. If this
 * input is not set, then the component will render as `default`
 *
 * `opened` is a flag that will dictate whether the drawer is opened or
 * closed. This value defaults to `false` and is generally left off when
 * instantiating the component.
 *
 * `fixedOpen` is a flag that can can be added to the component instatiation in
 * a similar manner to `opened`. This flag will tell the drawer to never
 * collapse down to a minified state, and is inteded for use in specific
 * applications where the design necessitates an 'always open' drawer.
 *
 */
class NavigationDrawerComponent {
    /**
     * PRIVATE
     *
     * Is the drawer currently being dragged around?
     *
     * @ignore
     */
    #isDragging;
    /**
     * PRIVATE
     *
     * The starting width for the drawer. Ostensibly used in dragging, but
     * uncertain if this is still required. Code has moved away from using JS
     * animation and towards capitalizing on Material's `opened` property and
     * premade transforms.
     *
     * @todo Ascertain the above and remove if necessary.
     *
     * @ignore
     */
    #startWidth;
    /**
     * PRIVATE
     *
     * The starting X-position for the drawer. Ostensibly used in dragging, but
     * uncertain if this is still required. Code has moved away from using JS
     * animation and towards capitalizing on Material's `opened` property and
     * premade transforms.
     *
     * @todo Ascertain the above and remove if necessary.
     *
     * @ignore
     */
    #startX;
    /*****************************************************************************
                                   CONSTRUCTOR
    *****************************************************************************/
    constructor(changeDetectorRef, router, media) {
        this.changeDetectorRef = changeDetectorRef;
        this.router = router;
        /**
         * Sets the width for the drawer when in its collapsed state.
         *
         * Leaving this here, as we may have use for it at some point, but let's make
         * it an undocumented Input for now.
         *
         * @ignore
         */
        this.collapsedWidth = '80px';
        /**
         * Sets the width for the drawer when in its expanded state.
         *
         * Leaving this here, as we may have use for it at some point, but let's make
         * it an undocumented Input for now.
         *
         * @ignore
         */
        this.expandedWidth = '360px';
        /**
         * An input flag denoting whether or not the Nav Drawer should never
         * collapse down to a minified state, and is intended for use in specific
         * applications where the design necessitates an 'always open' drawer.
         */
        this.fixedOpen = false;
        /**
         * An input flag denoting whether or not the Nav Drawer should show a minified
         * (thin) presentation when collapsed
         */
        this.minifyOnCollapse = false;
        /**
         * Menu items for the navdrawer
         */
        this.navDrawerMenuItems = MockSidebarMenuData;
        /**
         * An input flag denoting whether or not the Navigation Drawer presents in its
         * DEFAULT configuration (no branding, assuming the presence of an Application
         * Top Bar component on the page), it's STANDALONE configuration (with
         * branding), or as a MODAL.
         */
        this.navDrawerPresentation = NavDrawerPresentationEnum.DEFAULT;
        /**
         * A direct map to Angular Material's `opened` property of the `Side Nav`
         * component which this drawer is built on top of. Setting `opened` to `true`
         * or `false` expands and contracts the drawer, respectively, using the CSS
         * transforms that Material created under the hood.
         *
         * It is advisable to use this property instead of custom javascript
         * animations to facilitate the expansion/collapse becase additional ARIA
         * properties and classes are predicated on this component state marker.
         */
        this.opened = false;
        this.openedChange = new EventEmitter(); // Notifies parent of changes
        this.routeRequested = new EventEmitter();
        /*****************************************************************************
                                      PROPERTIES
      *****************************************************************************/
        /**
         * When not in MODAL mode, we want the drawer to collapse on mouse-out and on
         * selection of a menu item... UNLESS the drawer has already been manually
         * dragged open. We count such a state as an "intentional open" and respect
         * that the user deliberately expanded the drawer. This flag provides us a
         * check for such cases, and is used in conditional logic within the template.
         */
        this.intentionalOpen = false;
        /**
         * PRIVATE
         *
         * Is the drawer currently being dragged around?
         *
         * @ignore
         */
        this.#isDragging = false;
        /**
         * Determines if a menu item was clicked. Not sure we need this one.
         *
         * @ignore
         */
        this.isMenuItemClicked = false;
        /**
         * PRIVATE
         *
         * The starting width for the drawer. Ostensibly used in dragging, but
         * uncertain if this is still required. Code has moved away from using JS
         * animation and towards capitalizing on Material's `opened` property and
         * premade transforms.
         *
         * @todo Ascertain the above and remove if necessary.
         *
         * @ignore
         */
        this.#startWidth = 0;
        /**
         * PRIVATE
         *
         * The starting X-position for the drawer. Ostensibly used in dragging, but
         * uncertain if this is still required. Code has moved away from using JS
         * animation and towards capitalizing on Material's `opened` property and
         * premade transforms.
         *
         * @todo Ascertain the above and remove if necessary.
         *
         * @ignore
         */
        this.#startX = 0;
        /**
         * Since our `NavDrawerPresentationEnum` is used for evaluations not only in
         * our class code, but also in our template, we need to expose it publicly on
         * our class.
         */
        this.NavDrawerPresentationEnum = NavDrawerPresentationEnum;
        this.mobileQuery = media.matchMedia('(max-width: 600px)');
        this._mobileQueryListener = () => changeDetectorRef.detectChanges();
        this.mobileQuery.addListener(this._mobileQueryListener);
        this.boundDrag = this.drag.bind(this);
        this.boundStopDrag = this.stopDragging.bind(this);
    }
    ngOnInit() {
        this.setFirstItemActive();
    }
    setFirstItemActive() {
        if (this.navDrawerMenuItems && this.navDrawerMenuItems.length > 0) {
            const firstItem = this.navDrawerMenuItems[0];
            if (firstItem.sectionMenuItems && firstItem.sectionMenuItems.length > 0) {
                this.onMenuItemClick(firstItem.sectionMenuItems[0]);
                // The following check for sidenavMode is commented out because we have decided
                // to enable backdrop clicks for both modal and standalone presentations.
                // As a result, the distinction between 'over' and other modes is no longer necessary.
                // if(this.sidenavMode !== 'over') { //Commented
                // this.opened = false;
                // }
            }
        }
    }
    /*****************************************************************************
                                       METHODS
    *****************************************************************************/
    /**
     * Starts dragging, adds necessary event listeners, and appropriately sets
     * the relevant flags.
     */
    startDragging(event) {
        // Ensure only the left mouse button starts the drag
        if (event instanceof MouseEvent) {
            if (event.buttons !== 1) {
                console.log("Dragging only allowed with left mouse button.");
                return;
            }
            event.preventDefault(); // Prevents accidental text selection
        }
        if (this.#isDragging)
            return; // Prevent multiple registrations
        this.#isDragging = true;
        this.#startX =
            event instanceof MouseEvent ? event.clientX : event.touches[0].clientX;
        this.#startWidth = this.sidenavElement.nativeElement.offsetWidth;
        document.addEventListener("mousemove", this.boundDrag);
        document.addEventListener("touchmove", this.boundDrag);
        document.addEventListener("mouseup", this.boundStopDrag);
        document.addEventListener("mouseleave", this.boundStopDrag);
        document.addEventListener("touchend", this.boundStopDrag);
    }
    /**
     * Facilitates the snapping of the drag when the user drags past a certain
     * point.
     *
     * @todo This could probably be looked at a bit more closely and maybe cleaned up a bit.
     */
    drag(event) {
        if (!this.#isDragging)
            return;
        // Fix: If mouse button is released but `mouseup` didn’t trigger, stop dragging
        if (event instanceof MouseEvent && event.buttons === 0) {
            this.stopDragging();
            return;
        }
        const currentX = event instanceof MouseEvent ? event.clientX : event.touches[0].clientX;
        if (currentX <= parseInt(this.expandedWidth) - 100) {
            this.opened = false;
            this.openedChange.emit(this.opened);
            this.isMenuItemClicked = false;
        }
        else if (currentX >= parseInt(this.expandedWidth)) {
            this.opened = true;
            this.openedChange.emit(this.opened);
        }
    }
    /**
     * Stops dragging, removes necessary event listeners, and appropriately sets
     * the relevant flags.
     */
    stopDragging() {
        if (!this.#isDragging)
            return;
        this.#isDragging = false;
        document.removeEventListener("mousemove", this.boundDrag);
        document.removeEventListener("touchmove", this.boundDrag);
        document.removeEventListener("mouseup", this.boundStopDrag);
        document.removeEventListener("mouseleave", this.boundStopDrag);
        document.removeEventListener("touchend", this.boundStopDrag);
        const currentWidth = this.sidenavElement.nativeElement.offsetWidth;
        const collapsedWidth = parseInt(this.collapsedWidth);
        const expandedWidth = parseInt(this.expandedWidth);
        if (currentWidth > collapsedWidth) {
            this.intentionalOpen = true;
            this.opened = true;
            this.openedChange.emit(this.opened);
        }
        else if (currentWidth < expandedWidth) {
            this.intentionalOpen = false;
            this.opened = false;
            this.openedChange.emit(this.opened);
        }
    }
    /**
     * Handles removal of event listeners on destruction of a component instance.
     */
    ngOnDestroy() {
        this.mobileQuery.removeListener(this._mobileQueryListener);
    }
    /**
     * Do we actually need this, or is it just there to satisfy storybook/QA?
     *
     * I can't think offhand My tof a scenario in which we would want to reset all
     * toggled states if a menu item is selected...
     *
     * On speaking with Rahul, it also sounds as though we also only want the
     * drawer to collapse when an item is selected IF the user did not manually
     * open the drawer via drag.
     *
     * @param item
     */
    onMenuItemClick(item) {
        this.isMenuItemClicked = true;
        // this.intentionalOpen = true;
        // this.opened = true;
        // this.openedChange.emit(this.opened);
        // Reset toggled state for all items including child items
        const resetToggledState = (menuItems) => {
            menuItems.forEach((menuItem) => {
                if (menuItem.sectionMenuItems) {
                    resetToggledState(menuItem.sectionMenuItems);
                }
                else {
                    menuItem.toggled = false;
                }
                if (menuItem.children) {
                    resetToggledState(menuItem.children);
                }
            });
        };
        resetToggledState(this.navDrawerMenuItems);
        item.toggled = true;
    }
    /**
     * Toggles the expanded/collapsed state of a menu item that contains children
     *
     * @param item
     */
    toggleSubMenu(item) {
        if (item.children) {
            item.expanded = !item.expanded;
        }
    }
    routerClicked(route, item) {
        if (route) {
            // Navigate internally using Angular Router
            this.router.navigate([route]).catch(err => {
                console.error('Navigation error:', err);
                // Optionally emit event if navigation fails
                this.routeRequested.emit({ route, item });
            });
        }
        else {
            // No route provided, emit event for consumer to handle routing
            this.routeRequested.emit({ route: undefined, item });
        }
    }
    /*****************************************************************************
                                      ACCESSORS
    *****************************************************************************/
    /**
     * Returns the sidenav mode with respect to underlying content. (OVER, SIDE,
     * or PUSH)
     */
    get sidenavMode() {
        return this.mobileQuery.matches ||
            this.navDrawerPresentation === NavDrawerPresentationEnum.MODAL
            ? SideNavMode.OVER
            : SideNavMode.SIDE;
    }
    /**
     * Returns a boolean denoting if a given presentation of the Nav Drawer is
     * draggable. To arrive at this boolean, we look at both the presentation enum
     * and a whether or not the component is delcared as `fixedOpen`.
     */
    get isDraggable() {
        return (this.navDrawerPresentation !== NavDrawerPresentationEnum.MODAL &&
            !this.fixedOpen);
    }
    onBackdropClick() {
        this.opened = false; // Close the drawer
        this.openedChange.emit(this.opened); // Notify parent about the state change
        this.changeDetectorRef.detectChanges();
    }
    onMouseEnter() {
        if (!this.fixedOpen) {
            this.opened = true;
            this.openedChange.emit(this.opened);
            this.changeDetectorRef.detectChanges();
        }
    }
    onMouseLeave() {
        if (!this.fixedOpen && this.opened && !this.intentionalOpen) {
            this.opened = false;
            this.openedChange.emit(this.opened);
            this.changeDetectorRef.detectChanges();
        }
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "20.3.6", ngImport: i0, type: NavigationDrawerComponent, deps: [{ token: i0.ChangeDetectorRef }, { token: i1.Router }, { token: i2.MediaMatcher }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "17.0.0", version: "20.3.6", type: NavigationDrawerComponent, isStandalone: true, selector: "ba-navigation-drawer", inputs: { appLogo: "appLogo", appName: "appName", homeRoute: "homeRoute", collapsedWidth: "collapsedWidth", expandedWidth: "expandedWidth", fixedOpen: ["fixedOpen", "fixedOpen", coerceBooleanProperty], minifyOnCollapse: ["minifyOnCollapse", "minifyOnCollapse", coerceBooleanProperty], navBarBottomComponent1: "navBarBottomComponent1", navBarBottomComponent2: "navBarBottomComponent2", navBarBottomComponent3: "navBarBottomComponent3", navBarBottomComponent4: "navBarBottomComponent4", navDrawerMenuItems: "navDrawerMenuItems", navDrawerPresentation: "navDrawerPresentation", opened: ["opened", "opened", coerceBooleanProperty] }, outputs: { openedChange: "openedChange", routeRequested: "routeRequested" }, viewQueries: [{ propertyName: "sidenavElement", first: true, predicate: ["sidenav"], descendants: true, read: ElementRef }], ngImport: i0, template: "<mat-sidenav-container\n  autosize=\"true\"\n  color=\"primary\"\n  [hasBackdrop]=\"navDrawerPresentation === 'modal'\"\n  [class.collapse-to-rail] = \"minifyOnCollapse\"\n  (backdropClick)=\"onBackdropClick()\"\n>\n  <mat-sidenav\n    #sidenav\n    (mouseenter)=\"onMouseEnter()\"\n    (mouseleave)=\"onMouseLeave()\"\n    [mode]=\"sidenavMode\"\n    [opened]=\"fixedOpen ? true : opened\"\n    [autoFocus]=\"true\"\n  >\n    <!-- OPTIONAL BRANDING HEADER -->\n    <ng-container>\n      @if (appLogo) {\n        <div class=\"sidenav-header\">\n          <a class=\"app-home-link\" (click)=\"routerClicked(homeRoute)\">\n            @if (appLogo && opened) {\n              <img [src]=\"'./assets/' + appLogo + '-lettermark_Navigation_dark.svg'\" class=\"navigation__app-icon--dark letter-mark\" alt=\"dark App Logo\">\n            }\n            @if (appLogo && opened) {\n              <img [src]=\"'./assets/' + appLogo + '-lettermark_Navigation_light.svg'\" class=\"navigation__app-icon--light letter-mark\" alt=\"light App Logo\">\n            }\n            @if (appLogo && !opened) {\n              <img [src]=\"'./assets/' + appLogo + '-logomark_Navigation_dark.svg'\" class=\"navigation__app-icon--dark\" alt=\"dark App Logo\">\n            }\n            @if (appLogo && !opened) {\n              <img [src]=\"'./assets/' + appLogo + '-logomark_Navigation_light.svg'\" class=\"navigation__app-icon--light\" alt=\"light App Logo\">\n            }\n            @if (opened) {\n              <mat-divider\n                [vertical]=\"true\"\n                class=\"divider\"\n              ></mat-divider>\n            }\n            @if (appName) {\n              <h5 class=\"app-name\" [class.hide-app-name]=\"!opened\">\n                {{ appName }}\n              </h5>\n            }\n          </a>\n        </div>\n      }\n    </ng-container>\n\n    <!-- DYNAMIC MENU -->\n    @for (section of navDrawerMenuItems; track section) {\n      <ng-container>\n        <!-- header section -->\n        @if (section.sectionHeader) {\n          <div class=\"section-header\">\n            <!-- If collapsed, show an icon... -->\n            @if (section.sectionIcon && !opened && navDrawerPresentation !== NavDrawerPresentationEnum.MODAL) {\n              <mat-icon\n              matListItemIcon\n              [svgIcon]=\"'icon-'+section.sectionIcon\"\n              color=\"primary\"\n            ></mat-icon>\n            }\n            <!-- If expanded, show the section header text... -->\n            @if (opened) {\n              <h6 class=\"section-header-label\">\n                {{ section.sectionHeader }}\n              </h6>\n            }\n          </div>\n        }\n\n        <!-- navigation list -->\n        <mat-nav-list>\n          @for (item of section.sectionMenuItems; track item) {\n            <ng-container>\n              <mat-list-item\n                matListItemTitle\n                (mouseenter)=\"opened = true\"\n                (focus)=\"opened = true\"\n                (click)=\"onMenuItemClick(item); \n                toggleSubMenu(item); routerClicked(item.route, item)\"\n                [ngClass]=\"{ 'mat-list-item--toggle-on': item.toggled }\"\n                >\n                @if (item.icon) {\n                  <mat-icon matListItemIcon\n                    [svgIcon]=\"item.toggled ? 'icon-'+item.icon+'-filled' : 'icon-'+item.icon\"\n                  ></mat-icon>\n                }\n                @if (opened || navDrawerPresentation === NavDrawerPresentationEnum.MODAL) {\n                  <div matListItemTitle class=\"section-label\">\n                    {{ item.label }}\n                  </div>\n                }\n                @if (item.children && opened) {\n                  <mat-icon\n                    matListItemIcon\n                    class=\"dropdown-icon\"\n                    [class.expanded]=\"item.expanded\"\n                    svgIcon=\"icon-arrow-drop-down\"\n                  ></mat-icon>\n                }\n              </mat-list-item>\n\n              @if (item.children && item.expanded && opened) {\n                <div>\n                  <mat-nav-list class=\"level2-nav\">\n                    @for (child of item.children; track child) {\n                      <mat-list-item\n                        (click)=\"onMenuItemClick(child); routerClicked(child.route, child)\"\n                        [ngClass]=\"{ 'mat-list-item--toggle-on': child.toggled }\"\n                        >\n                        @if (child.icon) {\n                          <mat-icon matListItemIcon\n                            [svgIcon]=\"child.toggled ? 'icon-'+child.icon+'-filled' : 'icon-'+child.icon\"\n                          ></mat-icon>\n                        }\n                        <div matListItemTitle class=\"section-label\">\n                          {{ child.label }}\n                        </div>\n                      </mat-list-item>\n                    }\n                  </mat-nav-list>\n                </div>\n              }\n            </ng-container>\n          }\n          <mat-divider></mat-divider>\n        </mat-nav-list>\n      </ng-container>\n    }\n\n    <!-- BOTTOM BUTTONS (max: 4) -->\n    <!-- list for notification and profile (and up to two more)-->\n    <mat-nav-list class=\"bottom-icons\">\n      <ng-template *ngTemplateOutlet=\"navBarBottomComponent1\" />\n      <ng-template *ngTemplateOutlet=\"navBarBottomComponent2\" />\n      <ng-template *ngTemplateOutlet=\"navBarBottomComponent3\" />\n      <ng-template *ngTemplateOutlet=\"navBarBottomComponent4\" />\n    </mat-nav-list>\n\n    <!-- Drag bar -->\n    <!-- Removed Mat-divider as backdrop-click is not active for all type of navigation-drawer.\n    Kept the html code here for later improvements required in future-->\n    <mat-divider\n      vertical\n      class=\"sidenav-drag-handle\"\n      [class.enabled] = \"isDraggable\"\n      (mousedown)=\"isDraggable && startDragging($event)\"\n      (touchstart)=\"isDraggable && startDragging($event)\"\n      >\n    </mat-divider>\n  </mat-sidenav>\n  <mat-sidenav-content #sidenavContent class=\"content\">\n    <router-outlet></router-outlet>\n  </mat-sidenav-content>\n</mat-sidenav-container>\n", styles: ["mat-sidenav-container{height:100vh}.collapse-to-rail .mat-drawer{visibility:visible!important}.collapse-to-rail .mat-drawer[style*=\"visibility: hidden\"]{display:block!important;visibility:visible!important}.collapse-to-rail mat-sidenav:not(.mat-drawer-opened){transform:translateZ(0)!important;visibility:visible!important;width:80px!important;overflow:hidden}.collapse-to-rail .mat-drawer:not(.mat-drawer-opened):not(.mat-drawer-animating){visibility:visible!important;box-shadow:none!important}.collapse-to-rail .mat-drawer:not(.mat-drawer-opened):not(.mat-drawer-animating) .mat-drawer-inner-container{display:block!important}mat-sidenav{display:flex;flex-direction:column;height:100%;width:360px;box-shadow:none;border-right-color:var(--mat-sidenav-container-divider-color, transparent);border-right-width:1px;border-right-style:solid;-webkit-backdrop-filter:blur(100px);backdrop-filter:blur(100px);padding:20px 12px}mat-sidenav mat-list-item .mdc-list-item__content{overflow:visible}mat-sidenav mat-list-item.mat-mdc-list-item{background-color:transparent}mat-sidenav mat-list-item.mat-mdc-list-item:focus{background-color:var(--color-interactive-primary-background-enabled-tonal)}mat-sidenav mat-list-item.mat-mdc-list-item:active{background-color:var(--color-interactive-primary-background-selected)}mat-sidenav mat-list-item.mat-mdc-list-item:hover:not(:disabled),mat-sidenav mat-list-item.mat-mdc-list-item.mat-list-item--toggle-on{background-color:var(--color-interactive-primary-background-hovered)}mat-sidenav mat-list-item.mat-mdc-list-item:hover:not(:disabled) .mdc-list-item__primary-text,mat-sidenav mat-list-item.mat-mdc-list-item.mat-list-item--toggle-on .mdc-list-item__primary-text{text-shadow:var(--box-shadow-interactive-primary-glow-backlit);overflow:visible}mat-sidenav mat-list-item.mat-mdc-list-item:hover:not(:disabled) .mat-icon.mat-mdc-list-item-icon,mat-sidenav mat-list-item.mat-mdc-list-item.mat-list-item--toggle-on .mat-icon.mat-mdc-list-item-icon{overflow:visible;filter:drop-shadow(var(--box-shadow-interactive-primary-glow-backlit))}mat-sidenav mat-list-item .mdc-list-item__primary-text{font-weight:500}mat-sidenav .mat-mdc-nav-list{padding:0}mat-sidenav ba-user-profile{display:flex;justify-content:center;align-items:center}html .theme-condensed .mat-sidenav{padding:16px 12px}html .mat-sidenav{--mat-sidenav-container-width: 360px}html .mat-sidenav .mdc-list-item--with-leading-icon.mdc-list-item{padding-left:0;padding-right:0}html .mat-sidenav .level2-nav .mat-list-item.mat-mdc-list-item,html .mat-sidenav .level2-nav .mdc-list-item--with-leading-icon.mdc-list-item,html .mat-sidenav .level3-nav .mat-list-item.mat-mdc-list-item,html .mat-sidenav .level3-nav .mdc-list-item--with-leading-icon.mdc-list-item{padding-left:32px}.sidenav-header{justify-content:left;display:flex}.sidenav-header .divider{transform:rotate(36deg);margin-left:16px;margin-right:16px}.sidenav-header .app-home-link{justify-content:left;display:flex}.sidenav-header .app-home-link:focus{outline:none}.app-name{color:var(--color-ui-textandicon-high);display:flex;align-items:center}.hide-app-name{opacity:0px;height:0px;display:none}.navigation__app-icon--light{display:none}.navigation__app-icon--dark{display:block}.navigation__app-icon--dark.letter-mark{padding-left:16px}.theme-light .navigation__app-icon--light{display:block}.theme-light .navigation__app-icon--light.letter-mark{padding-left:16px}.theme-light .navigation__app-icon--dark{display:none}.section-header{justify-content:left;text-align:center;align-items:center}.section-header .section-header-label{font-size:14px;padding-left:16px;color:var(--color-ui-textandicon-low);text-align:left}.bottom-icons{display:flex;flex-shrink:0;position:absolute;bottom:12px;flex-direction:column;align-items:center;justify-content:flex-end;padding:0 8px!important}.dropdown-icon{position:absolute;right:0;transition:transform .3s}mat-nav-list{flex:1 1 auto;overflow-y:auto}.dropdown-icon.expanded{transform:rotate(180deg)}.section-label{max-width:82%}.mat-sidenav .mat-divider{margin-top:12px;margin-bottom:12px}.mat-sidenav .mat-divider.sidenav-drag-handle{position:absolute;height:100%;top:0;right:0;bottom:0;margin:0;padding-left:12px}.mat-sidenav .mat-divider.sidenav-drag-handle.enabled{cursor:ew-resize}.mat-sidenav .mat-divider.sidenav-drag-handle.enabled:hover,.mat-sidenav .mat-divider.sidenav-drag-handle.enabled:focus,.mat-sidenav .mat-divider.sidenav-drag-handle.enabled:active{border-right-width:4px}.mat-sidenav .mat-divider.sidenav-drag-handle.enabled:hover{border-color:var(--color-interactive-accent-border-hovered)}.mat-sidenav .mat-divider.sidenav-drag-handle.enabled:focus{border-color:var(--color-interactive-accent-border-selected)}.mat-sidenav .mat-divider.sidenav-drag-handle.enabled:active{border-color:var(--color-interactive-accent-border-enabled)}\n"], dependencies: [{ kind: "ngmodule", type: CommonModule }, { kind: "directive", type: i3.NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }, { kind: "directive", type: i3.NgTemplateOutlet, selector: "[ngTemplateOutlet]", inputs: ["ngTemplateOutletContext", "ngTemplateOutlet", "ngTemplateOutletInjector"] }, { kind: "ngmodule", type: MatToolbarModule }, { kind: "ngmodule", type: MatSidenavModule }, { kind: "component", type: i4.MatSidenav, selector: "mat-sidenav", inputs: ["fixedInViewport", "fixedTopGap", "fixedBottomGap"], exportAs: ["matSidenav"] }, { kind: "component", type: i4.MatSidenavContainer, selector: "mat-sidenav-container", exportAs: ["matSidenavContainer"] }, { kind: "component", type: i4.MatSidenavContent, selector: "mat-sidenav-content" }, { kind: "ngmodule", type: MatButtonModule }, { kind: "ngmodule", type: MatIconModule }, { kind: "component", type: i5.MatIcon, selector: "mat-icon", inputs: ["color", "inline", "svgIcon", "fontSet", "fontIcon"], exportAs: ["matIcon"] }, { kind: "directive", type: RouterOutlet, selector: "router-outlet", inputs: ["name", "routerOutletData"], outputs: ["activate", "deactivate", "attach", "detach"], exportAs: ["outlet"] }, { kind: "ngmodule", type: RouterModule }, { kind: "ngmodule", type: MatListModule }, { kind: "component", type: i6.MatNavList, selector: "mat-nav-list", exportAs: ["matNavList"] }, { kind: "component", type: i6.MatListItem, selector: "mat-list-item, a[mat-list-item], button[mat-list-item]", inputs: ["activated"], exportAs: ["matListItem"] }, { kind: "directive", type: i6.MatListItemIcon, selector: "[matListItemIcon]" }, { kind: "component", type: i6.MatDivider, selector: "mat-divider", inputs: ["vertical", "inset"] }, { kind: "directive", type: i6.MatListItemTitle, selector: "[matListItemTitle]" }, { kind: "ngmodule", type: MatDividerModule }, { kind: "ngmodule", type: MatMenuModule }, { kind: "ngmodule", type: IconRegistryModule }], encapsulation: i0.ViewEncapsulation.None }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "20.3.6", ngImport: i0, type: NavigationDrawerComponent, decorators: [{
            type: Component,
            args: [{ selector: 'ba-navigation-drawer', imports: [
                        CommonModule,
                        MatToolbarModule,
                        MatSidenavModule,
                        MatButtonModule,
                        MatIconModule,
                        RouterOutlet,
                        RouterModule,
                        MatListModule,
                        RouterModule,
                        MatNavList,
                        MatDividerModule,
                        MatMenuModule,
                        MatSidenavModule,
                        IconRegistryModule
                    ], encapsulation: ViewEncapsulation.None, template: "<mat-sidenav-container\n  autosize=\"true\"\n  color=\"primary\"\n  [hasBackdrop]=\"navDrawerPresentation === 'modal'\"\n  [class.collapse-to-rail] = \"minifyOnCollapse\"\n  (backdropClick)=\"onBackdropClick()\"\n>\n  <mat-sidenav\n    #sidenav\n    (mouseenter)=\"onMouseEnter()\"\n    (mouseleave)=\"onMouseLeave()\"\n    [mode]=\"sidenavMode\"\n    [opened]=\"fixedOpen ? true : opened\"\n    [autoFocus]=\"true\"\n  >\n    <!-- OPTIONAL BRANDING HEADER -->\n    <ng-container>\n      @if (appLogo) {\n        <div class=\"sidenav-header\">\n          <a class=\"app-home-link\" (click)=\"routerClicked(homeRoute)\">\n            @if (appLogo && opened) {\n              <img [src]=\"'./assets/' + appLogo + '-lettermark_Navigation_dark.svg'\" class=\"navigation__app-icon--dark letter-mark\" alt=\"dark App Logo\">\n            }\n            @if (appLogo && opened) {\n              <img [src]=\"'./assets/' + appLogo + '-lettermark_Navigation_light.svg'\" class=\"navigation__app-icon--light letter-mark\" alt=\"light App Logo\">\n            }\n            @if (appLogo && !opened) {\n              <img [src]=\"'./assets/' + appLogo + '-logomark_Navigation_dark.svg'\" class=\"navigation__app-icon--dark\" alt=\"dark App Logo\">\n            }\n            @if (appLogo && !opened) {\n              <img [src]=\"'./assets/' + appLogo + '-logomark_Navigation_light.svg'\" class=\"navigation__app-icon--light\" alt=\"light App Logo\">\n            }\n            @if (opened) {\n              <mat-divider\n                [vertical]=\"true\"\n                class=\"divider\"\n              ></mat-divider>\n            }\n            @if (appName) {\n              <h5 class=\"app-name\" [class.hide-app-name]=\"!opened\">\n                {{ appName }}\n              </h5>\n            }\n          </a>\n        </div>\n      }\n    </ng-container>\n\n    <!-- DYNAMIC MENU -->\n    @for (section of navDrawerMenuItems; track section) {\n      <ng-container>\n        <!-- header section -->\n        @if (section.sectionHeader) {\n          <div class=\"section-header\">\n            <!-- If collapsed, show an icon... -->\n            @if (section.sectionIcon && !opened && navDrawerPresentation !== NavDrawerPresentationEnum.MODAL) {\n              <mat-icon\n              matListItemIcon\n              [svgIcon]=\"'icon-'+section.sectionIcon\"\n              color=\"primary\"\n            ></mat-icon>\n            }\n            <!-- If expanded, show the section header text... -->\n            @if (opened) {\n              <h6 class=\"section-header-label\">\n                {{ section.sectionHeader }}\n              </h6>\n            }\n          </div>\n        }\n\n        <!-- navigation list -->\n        <mat-nav-list>\n          @for (item of section.sectionMenuItems; track item) {\n            <ng-container>\n              <mat-list-item\n                matListItemTitle\n                (mouseenter)=\"opened = true\"\n                (focus)=\"opened = true\"\n                (click)=\"onMenuItemClick(item); \n                toggleSubMenu(item); routerClicked(item.route, item)\"\n                [ngClass]=\"{ 'mat-list-item--toggle-on': item.toggled }\"\n                >\n                @if (item.icon) {\n                  <mat-icon matListItemIcon\n                    [svgIcon]=\"item.toggled ? 'icon-'+item.icon+'-filled' : 'icon-'+item.icon\"\n                  ></mat-icon>\n                }\n                @if (opened || navDrawerPresentation === NavDrawerPresentationEnum.MODAL) {\n                  <div matListItemTitle class=\"section-label\">\n                    {{ item.label }}\n                  </div>\n                }\n                @if (item.children && opened) {\n                  <mat-icon\n                    matListItemIcon\n                    class=\"dropdown-icon\"\n                    [class.expanded]=\"item.expanded\"\n                    svgIcon=\"icon-arrow-drop-down\"\n                  ></mat-icon>\n                }\n              </mat-list-item>\n\n              @if (item.children && item.expanded && opened) {\n                <div>\n                  <mat-nav-list class=\"level2-nav\">\n                    @for (child of item.children; track child) {\n                      <mat-list-item\n                        (click)=\"onMenuItemClick(child); routerClicked(child.route, child)\"\n                        [ngClass]=\"{ 'mat-list-item--toggle-on': child.toggled }\"\n                        >\n                        @if (child.icon) {\n                          <mat-icon matListItemIcon\n                            [svgIcon]=\"child.toggled ? 'icon-'+child.icon+'-filled' : 'icon-'+child.icon\"\n                          ></mat-icon>\n                        }\n                        <div matListItemTitle class=\"section-label\">\n                          {{ child.label }}\n                        </div>\n                      </mat-list-item>\n                    }\n                  </mat-nav-list>\n                </div>\n              }\n            </ng-container>\n          }\n          <mat-divider></mat-divider>\n        </mat-nav-list>\n      </ng-container>\n    }\n\n    <!-- BOTTOM BUTTONS (max: 4) -->\n    <!-- list for notification and profile (and up to two more)-->\n    <mat-nav-list class=\"bottom-icons\">\n      <ng-template *ngTemplateOutlet=\"navBarBottomComponent1\" />\n      <ng-template *ngTemplateOutlet=\"navBarBottomComponent2\" />\n      <ng-template *ngTemplateOutlet=\"navBarBottomComponent3\" />\n      <ng-template *ngTemplateOutlet=\"navBarBottomComponent4\" />\n    </mat-nav-list>\n\n    <!-- Drag bar -->\n    <!-- Removed Mat-divider as backdrop-click is not active for all type of navigation-drawer.\n    Kept the html code here for later improvements required in future-->\n    <mat-divider\n      vertical\n      class=\"sidenav-drag-handle\"\n      [class.enabled] = \"isDraggable\"\n      (mousedown)=\"isDraggable && startDragging($event)\"\n      (touchstart)=\"isDraggable && startDragging($event)\"\n      >\n    </mat-divider>\n  </mat-sidenav>\n  <mat-sidenav-content #sidenavContent class=\"content\">\n    <router-outlet></router-outlet>\n  </mat-sidenav-content>\n</mat-sidenav-container>\n", styles: ["mat-sidenav-container{height:100vh}.collapse-to-rail .mat-drawer{visibility:visible!important}.collapse-to-rail .mat-drawer[style*=\"visibility: hidden\"]{display:block!important;visibility:visible!important}.collapse-to-rail mat-sidenav:not(.mat-drawer-opened){transform:translateZ(0)!important;visibility:visible!important;width:80px!important;overflow:hidden}.collapse-to-rail .mat-drawer:not(.mat-drawer-opened):not(.mat-drawer-animating){visibility:visible!important;box-shadow:none!important}.collapse-to-rail .mat-drawer:not(.mat-drawer-opened):not(.mat-drawer-animating) .mat-drawer-inner-container{display:block!important}mat-sidenav{display:flex;flex-direction:column;height:100%;width:360px;box-shadow:none;border-right-color:var(--mat-sidenav-container-divider-color, transparent);border-right-width:1px;border-right-style:solid;-webkit-backdrop-filter:blur(100px);backdrop-filter:blur(100px);padding:20px 12px}mat-sidenav mat-list-item .mdc-list-item__content{overflow:visible}mat-sidenav mat-list-item.mat-mdc-list-item{background-color:transparent}mat-sidenav mat-list-item.mat-mdc-list-item:focus{background-color:var(--color-interactive-primary-background-enabled-tonal)}mat-sidenav mat-list-item.mat-mdc-list-item:active{background-color:var(--color-interactive-primary-background-selected)}mat-sidenav mat-list-item.mat-mdc-list-item:hover:not(:disabled),mat-sidenav mat-list-item.mat-mdc-list-item.mat-list-item--toggle-on{background-color:var(--color-interactive-primary-background-hovered)}mat-sidenav mat-list-item.mat-mdc-list-item:hover:not(:disabled) .mdc-list-item__primary-text,mat-sidenav mat-list-item.mat-mdc-list-item.mat-list-item--toggle-on .mdc-list-item__primary-text{text-shadow:var(--box-shadow-interactive-primary-glow-backlit);overflow:visible}mat-sidenav mat-list-item.mat-mdc-list-item:hover:not(:disabled) .mat-icon.mat-mdc-list-item-icon,mat-sidenav mat-list-item.mat-mdc-list-item.mat-list-item--toggle-on .mat-icon.mat-mdc-list-item-icon{overflow:visible;filter:drop-shadow(var(--box-shadow-interactive-primary-glow-backlit))}mat-sidenav mat-list-item .mdc-list-item__primary-text{font-weight:500}mat-sidenav .mat-mdc-nav-list{padding:0}mat-sidenav ba-user-profile{display:flex;justify-content:center;align-items:center}html .theme-condensed .mat-sidenav{padding:16px 12px}html .mat-sidenav{--mat-sidenav-container-width: 360px}html .mat-sidenav .mdc-list-item--with-leading-icon.mdc-list-item{padding-left:0;padding-right:0}html .mat-sidenav .level2-nav .mat-list-item.mat-mdc-list-item,html .mat-sidenav .level2-nav .mdc-list-item--with-leading-icon.mdc-list-item,html .mat-sidenav .level3-nav .mat-list-item.mat-mdc-list-item,html .mat-sidenav .level3-nav .mdc-list-item--with-leading-icon.mdc-list-item{padding-left:32px}.sidenav-header{justify-content:left;display:flex}.sidenav-header .divider{transform:rotate(36deg);margin-left:16px;margin-right:16px}.sidenav-header .app-home-link{justify-content:left;display:flex}.sidenav-header .app-home-link:focus{outline:none}.app-name{color:var(--color-ui-textandicon-high);display:flex;align-items:center}.hide-app-name{opacity:0px;height:0px;display:none}.navigation__app-icon--light{display:none}.navigation__app-icon--dark{display:block}.navigation__app-icon--dark.letter-mark{padding-left:16px}.theme-light .navigation__app-icon--light{display:block}.theme-light .navigation__app-icon--light.letter-mark{padding-left:16px}.theme-light .navigation__app-icon--dark{display:none}.section-header{justify-content:left;text-align:center;align-items:center}.section-header .section-header-label{font-size:14px;padding-left:16px;color:var(--color-ui-textandicon-low);text-align:left}.bottom-icons{display:flex;flex-shrink:0;position:absolute;bottom:12px;flex-direction:column;align-items:center;justify-content:flex-end;padding:0 8px!important}.dropdown-icon{position:absolute;right:0;transition:transform .3s}mat-nav-list{flex:1 1 auto;overflow-y:auto}.dropdown-icon.expanded{transform:rotate(180deg)}.section-label{max-width:82%}.mat-sidenav .mat-divider{margin-top:12px;margin-bottom:12px}.mat-sidenav .mat-divider.sidenav-drag-handle{position:absolute;height:100%;top:0;right:0;bottom:0;margin:0;padding-left:12px}.mat-sidenav .mat-divider.sidenav-drag-handle.enabled{cursor:ew-resize}.mat-sidenav .mat-divider.sidenav-drag-handle.enabled:hover,.mat-sidenav .mat-divider.sidenav-drag-handle.enabled:focus,.mat-sidenav .mat-divider.sidenav-drag-handle.enabled:active{border-right-width:4px}.mat-sidenav .mat-divider.sidenav-drag-handle.enabled:hover{border-color:var(--color-interactive-accent-border-hovered)}.mat-sidenav .mat-divider.sidenav-drag-handle.enabled:focus{border-color:var(--color-interactive-accent-border-selected)}.mat-sidenav .mat-divider.sidenav-drag-handle.enabled:active{border-color:var(--color-interactive-accent-border-enabled)}\n"] }]
        }], ctorParameters: () => [{ type: i0.ChangeDetectorRef }, { type: i1.Router }, { type: i2.MediaMatcher }], propDecorators: { sidenavElement: [{
                type: ViewChild,
                args: ['sidenav', { read: ElementRef }]
            }], appLogo: [{
                type: Input
            }], appName: [{
                type: Input
            }], homeRoute: [{
                type: Input
            }], collapsedWidth: [{
                type: Input
            }], expandedWidth: [{
                type: Input
            }], fixedOpen: [{
                type: Input,
                args: [{ transform: coerceBooleanProperty }]
            }], minifyOnCollapse: [{
                type: Input,
                args: [{ transform: coerceBooleanProperty }]
            }], navBarBottomComponent1: [{
                type: Input
            }], navBarBottomComponent2: [{
                type: Input
            }], navBarBottomComponent3: [{
                type: Input
            }], navBarBottomComponent4: [{
                type: Input
            }], navDrawerMenuItems: [{
                type: Input
            }], navDrawerPresentation: [{
                type: Input
            }], opened: [{
                type: Input,
                args: [{ transform: coerceBooleanProperty }]
            }], openedChange: [{
                type: Output
            }], routeRequested: [{
                type: Output
            }] } });

/**
 * Generated bundle index. Do not edit.
 */

export { MockSidebarMenuData, NavDrawerPresentationEnum, NavigationDrawerComponent };
