// @jeppesen-foreflight/dp-dls-global-angular/fesm2022/jeppesen-foreflight-dp-dls-global-angular-user-profile.mjs
import * as i0 from '@angular/core';
import { Input, Component } from '@angular/core';
import { MatTabsModule } from '@angular/material/tabs';
import { MatIconModule } from '@angular/material/icon';
import { MatToolbarModule } from '@angular/material/toolbar';
import { MatButtonModule } from '@angular/material/button';
import * as i1 from '@angular/common';
import { CommonModule } from '@angular/common';
import { coerceBooleanProperty } from '@angular/cdk/coercion';

/**
 * The Atmosphere User Profile Component provides a means for an application to
 * display the currently logged-in user. It is configurable to use either an
 * image or a generic initial (similar in concept to MS Teams, Slack, etc.), and
 * is also configurable to be either clickable or static. In the clickable
 * presentation, a panel is opened allowing access to further controls such as
 * profile settings and logging out.
 *
 * At present, the component accepts three properties: the `clickable` flag,
 * a url/path for a user avatar image, and the user's first name. A possible
 * future state could condense the avatar image and first name inputs into a
 * single user object which could be used consistently between components.
 */
class UserProfileComponent {
    constructor() {
        /**
         * A flag denoting whether or not an instance of the User Profile component is
         * clickable or is static. When the `clickable` property is present, the
         * component will render in its actionable/clickable state.
         *
         */
        this.clickable = false;
        // I feel like there should be some way to infer ^this from the DOM. The
        // component contains the class `mat-mdc-menu-trigger` when it is set up to be
        // expandable. We should be able to use that directly, but I need to work out
        // the syntax.
        this.small = false;
        /**
         * A string for an image source to use in rendering the User Profile
         * component. If the string is either absent or fails a quick validation
         * check, then the component will default to the monogram presentation.
         */
        this.userAvatarSource = '';
    }
    /**
     * Returns a boolean denoting whether or not the string supplied in the
     * `imageSource` input represents an image file or not.
     */
    hasValidImageSource() {
        const imageExtensions = ['.jpg', '.jpeg', '.png', '.gif', '.bmp', '.svg'];
        return typeof this.userAvatarSource === 'string' &&
            (this.userAvatarSource.startsWith('http') || imageExtensions.some(ext => this.userAvatarSource.endsWith(ext)));
    }
    ;
    /**
     * Returns a single-initial monogram based on the logged-in user's first name.
     * THe first name is passed into the component as the `userFirstName` input.
     */
    getMonogram() {
        return this.userFirstName.charAt(0).toUpperCase();
    }
    getProfileClass() {
        if (this.clickable) {
            return 'user-profile--clickable';
        }
        else if (this.small) {
            return 'user-profile--display-only-small';
        }
        else {
            return 'user-profile--display-only';
        }
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "20.3.6", ngImport: i0, type: UserProfileComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "17.0.0", version: "20.3.6", type: UserProfileComponent, isStandalone: true, selector: "ba-user-profile", inputs: { clickable: ["clickable", "clickable", coerceBooleanProperty], small: ["small", "small", coerceBooleanProperty], userAvatarSource: "userAvatarSource", userFirstName: "userFirstName" }, ngImport: i0, template: "<ng-container>\n  <div class=\"user-profile\" [ngClass]=\"getProfileClass()\">\n    <div class=\"user-profile__content\">\n      @switch (hasValidImageSource()) {\n        @case (true) {\n          <img [src]=\"userAvatarSource\" alt=\"User profile image\" class=\"user-profile__image\" />\n        }\n        @case (false) {\n          <div class=\"user-profile__letter\">{{ getMonogram() }}</div>\n        }\n      }\n    </div>\n  </div>\n</ng-container>", styles: [".user-profile{display:flex;justify-content:center;align-items:center;border-radius:50%;overflow:hidden}.user-profile--display-only{width:48px;height:48px;font-size:22px;cursor:default}.user-profile--display-only-small{width:24px;height:24px;font-size:14px;cursor:default}.user-profile--clickable{width:40px;height:40px;font-size:14px;cursor:pointer}.user-profile__content{display:flex;justify-content:center;align-items:center;border-radius:50%;overflow:hidden;width:100%;height:100%}.user-profile--clickable .user-profile__content{width:32px;height:32px}.user-profile__image{width:100%;height:100%;object-fit:cover}.user-profile__letter{background-color:var(--color-static-gray);color:var(--color-static-white);font-family:Boeing Meso;font-weight:500;font-style:normal;display:flex;justify-content:center;align-items:center;width:100%;height:100%}\n"], dependencies: [{ kind: "ngmodule", type: MatTabsModule }, { kind: "ngmodule", type: MatIconModule }, { kind: "ngmodule", type: MatToolbarModule }, { kind: "ngmodule", type: MatButtonModule }, { kind: "ngmodule", type: CommonModule }, { kind: "directive", type: i1.NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "20.3.6", ngImport: i0, type: UserProfileComponent, decorators: [{
            type: Component,
            args: [{ selector: 'ba-user-profile', imports: [
                        MatTabsModule,
                        MatIconModule,
                        MatToolbarModule,
                        MatButtonModule,
                        CommonModule,
                        // IconRegistryModule
                    ], template: "<ng-container>\n  <div class=\"user-profile\" [ngClass]=\"getProfileClass()\">\n    <div class=\"user-profile__content\">\n      @switch (hasValidImageSource()) {\n        @case (true) {\n          <img [src]=\"userAvatarSource\" alt=\"User profile image\" class=\"user-profile__image\" />\n        }\n        @case (false) {\n          <div class=\"user-profile__letter\">{{ getMonogram() }}</div>\n        }\n      }\n    </div>\n  </div>\n</ng-container>", styles: [".user-profile{display:flex;justify-content:center;align-items:center;border-radius:50%;overflow:hidden}.user-profile--display-only{width:48px;height:48px;font-size:22px;cursor:default}.user-profile--display-only-small{width:24px;height:24px;font-size:14px;cursor:default}.user-profile--clickable{width:40px;height:40px;font-size:14px;cursor:pointer}.user-profile__content{display:flex;justify-content:center;align-items:center;border-radius:50%;overflow:hidden;width:100%;height:100%}.user-profile--clickable .user-profile__content{width:32px;height:32px}.user-profile__image{width:100%;height:100%;object-fit:cover}.user-profile__letter{background-color:var(--color-static-gray);color:var(--color-static-white);font-family:Boeing Meso;font-weight:500;font-style:normal;display:flex;justify-content:center;align-items:center;width:100%;height:100%}\n"] }]
        }], propDecorators: { clickable: [{
                type: Input,
                args: [{ transform: coerceBooleanProperty }]
            }], small: [{
                type: Input,
                args: [{ transform: coerceBooleanProperty }]
            }], userAvatarSource: [{
                type: Input
            }], userFirstName: [{
                type: Input,
                args: [{ required: true }]
            }] } });

/**
 * Generated bundle index. Do not edit.
 */

export { UserProfileComponent };
