// @jeppesen-foreflight/dp-dls-global-angular/fesm2022/jeppesen-foreflight-dp-dls-global-angular-theme-toggle.mjs
import * as i0 from '@angular/core';
import { InjectionToken, Injectable, DOCUMENT, PLATFORM_ID, Inject, Component, NgModule } from '@angular/core';
import { MatIconButton, MatButtonModule } from '@angular/material/button';
import { MatTooltip } from '@angular/material/tooltip';
import { MatIcon, MatIconModule } from '@angular/material/icon';
import { BehaviorSubject } from 'rxjs';
import { isPlatformBrowser } from '@angular/common';
import { IconRegistryModule } from '@jeppesen-foreflight/dp-dls-global-angular/icon-registry';

/**
 * Injection Token for ThemeStorage
 */
const THEME_STORAGE_SERVICE = new InjectionToken("THEME_STORAGE");
/**
 * Browser Local Storage based implementation of ThemeStorage
 */
class ThemeLocalStorageService {
    constructor() {
        /**
         * key for the Local Storage
         */
        this.LOCAL_STORAGE_KEY = "theme";
    }
    save(theme) {
        localStorage.setItem(this.LOCAL_STORAGE_KEY, theme.toString());
    }
    get() {
        return localStorage.getItem(this.LOCAL_STORAGE_KEY) || undefined;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "20.3.6", ngImport: i0, type: ThemeLocalStorageService, deps: [], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "20.3.6", ngImport: i0, type: ThemeLocalStorageService }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "20.3.6", ngImport: i0, type: ThemeLocalStorageService, decorators: [{
            type: Injectable
        }] });

/** enum for two Mode types */
var Theme;
(function (Theme) {
    /**
     * Light Mode
     */
    Theme["LIGHT"] = "theme-light";
    /**
     * Dark Mode
     */
    Theme["DARK"] = "theme-dark";
})(Theme || (Theme = {}));

/**
 * Angular service that provides the theme toggle feature.
 * In summary this service adds the `class='theme-light'` to the document.body element and
 * styles change based on the class added to the document.body
 *
 * Also provides a Observable that emits the current theme every time theme changes
 */
class ThemeToggleService {
    constructor(document, themeStorage, platformId) {
        this.document = document;
        this.themeStorage = themeStorage;
        this.platformId = platformId;
        /**
         * contains the current active theme
         * avoid mutating it directly, instead use `updateCurrentTheme`
         */
        this.currentTheme = Theme.DARK;
        /**
         * BehaviorSubject that detects the theme changes
         */
        this.themeChangedSubject = new BehaviorSubject(this.currentTheme);
        this.themeChanged$ = this.themeChangedSubject.asObservable();
        this.init(platformId);
    }
    /**
     * Function to mutate the currentTheme
     * @param theme Theme
     */
    updateCurrentTheme(theme, skipSave) {
        this.currentTheme = theme;
        this.themeChangedSubject.next(this.currentTheme);
        if (!skipSave) {
            this.themeStorage.save(this.currentTheme);
        }
    }
    /**
     * Init function that update the application based on the initial theme value
     * Flow as below
     * 1 - If there is a saved theme in the browser - use this as the initial value
     * 2 - If there is no saved theme, Check for the preferred device theme
     * 3 - If device theme is dark, set the init value to `dark`
     * 4 - Else set the default value to `light`
     */
    init(platformId) {
        let initTheme = Theme.DARK;
        if (isPlatformBrowser(platformId)) {
            initTheme = this.themeStorage.get();
            if (!initTheme) {
                const deviceMode = window.matchMedia("(prefers-color-scheme: light)");
                initTheme = deviceMode.matches ? Theme.LIGHT : Theme.DARK;
            }
            this.updateCurrentTheme(initTheme);
        }
        else {
            this.updateCurrentTheme(initTheme, true);
        }
        this.document.body.classList.toggle(Theme.LIGHT);
        this.document.body.classList.toggle(Theme.DARK);
        this.document.body.classList.add(this.currentTheme);
    }
    /**
     * Function to get the current theme
     * @returns Current theme
     * Exposed publicly
     */
    getCurrentTheme() {
        return this.currentTheme;
    }
    /**
     * Function that toggles the theme
     * Exposed publicly
     */
    toggleTheme() {
        this.document.body.classList.toggle(Theme.LIGHT);
        this.document.body.classList.toggle(Theme.DARK);
        if (this.currentTheme === Theme.LIGHT) {
            this.updateCurrentTheme(Theme.DARK);
        }
        else {
            this.updateCurrentTheme(Theme.LIGHT);
        }
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "20.3.6", ngImport: i0, type: ThemeToggleService, deps: [{ token: DOCUMENT }, { token: THEME_STORAGE_SERVICE }, { token: PLATFORM_ID }], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "20.3.6", ngImport: i0, type: ThemeToggleService }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "20.3.6", ngImport: i0, type: ThemeToggleService, decorators: [{
            type: Injectable
        }], ctorParameters: () => [{ type: Document, decorators: [{
                    type: Inject,
                    args: [DOCUMENT]
                }] }, { type: undefined, decorators: [{
                    type: Inject,
                    args: [THEME_STORAGE_SERVICE]
                }] }, { type: undefined, decorators: [{
                    type: Inject,
                    args: [PLATFORM_ID]
                }] }] });

/**
 * Angular component to switch the Theme.
 * Also developers can create their own components with the use of `ThemeToggleService`
 * @example
 * ```
 * <ba-theme-toggle></ba-theme-toggle>
 * ```
 */
class ThemeToggleComponent {
    constructor(themeToggleService) {
        this.themeToggleService = themeToggleService;
    }
    toggle() {
        this.themeToggleService.toggleTheme();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "20.3.6", ngImport: i0, type: ThemeToggleComponent, deps: [{ token: ThemeToggleService }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "20.3.6", type: ThemeToggleComponent, isStandalone: true, selector: "ba-theme-toggle", ngImport: i0, template: "<button mat-icon-button matTooltip=\"Primary\" color=\"primary\" aria-label=\"Toggle theme\" (click)=\"toggle()\">\n  <mat-icon svgIcon=\"icon-dark-mode\"></mat-icon>\n</button>\n\n", styles: [""], dependencies: [{ kind: "component", type: MatIconButton, selector: "button[mat-icon-button], a[mat-icon-button], button[matIconButton], a[matIconButton]", exportAs: ["matButton", "matAnchor"] }, { kind: "directive", type: MatTooltip, selector: "[matTooltip]", inputs: ["matTooltipPosition", "matTooltipPositionAtOrigin", "matTooltipDisabled", "matTooltipShowDelay", "matTooltipHideDelay", "matTooltipTouchGestures", "matTooltip", "matTooltipClass"], exportAs: ["matTooltip"] }, { kind: "component", type: MatIcon, selector: "mat-icon", inputs: ["color", "inline", "svgIcon", "fontSet", "fontIcon"], exportAs: ["matIcon"] }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "20.3.6", ngImport: i0, type: ThemeToggleComponent, decorators: [{
            type: Component,
            args: [{ selector: "ba-theme-toggle", imports: [MatIconButton, MatTooltip, MatIcon], template: "<button mat-icon-button matTooltip=\"Primary\" color=\"primary\" aria-label=\"Toggle theme\" (click)=\"toggle()\">\n  <mat-icon svgIcon=\"icon-dark-mode\"></mat-icon>\n</button>\n\n" }]
        }], ctorParameters: () => [{ type: ThemeToggleService }] });

/**
 * Angular module for mode toggling feature
 * Contains
 *  * ThemeToggleComponent
 *  * ThemeToggleService
 */
class ThemeToggleModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "20.3.6", ngImport: i0, type: ThemeToggleModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "20.3.6", ngImport: i0, type: ThemeToggleModule, imports: [MatButtonModule, MatIconModule, IconRegistryModule, ThemeToggleComponent], exports: [ThemeToggleComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "20.3.6", ngImport: i0, type: ThemeToggleModule, providers: [
            ThemeToggleService,
            {
                provide: THEME_STORAGE_SERVICE,
                useClass: ThemeLocalStorageService,
            },
        ], imports: [MatButtonModule, MatIconModule, IconRegistryModule, ThemeToggleComponent] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "20.3.6", ngImport: i0, type: ThemeToggleModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [MatButtonModule, MatIconModule, IconRegistryModule, ThemeToggleComponent],
                    providers: [
                        ThemeToggleService,
                        {
                            provide: THEME_STORAGE_SERVICE,
                            useClass: ThemeLocalStorageService,
                        },
                    ],
                    declarations: [],
                    exports: [ThemeToggleComponent],
                }]
        }] });

/**
 * Generated bundle index. Do not edit.
 */

export { ThemeToggleComponent, ThemeToggleModule, ThemeToggleService };
