// @jeppesen-foreflight/dp-dls-global-angular/fesm2022/jeppesen-foreflight-dp-dls-global-angular-snackbar.mjs
import * as i0 from '@angular/core';
import { Input, Inject, Component } from '@angular/core';
import * as i2 from '@angular/material/button';
import { MatButtonModule } from '@angular/material/button';
import * as i3 from '@angular/material/icon';
import { MatIconModule } from '@angular/material/icon';
import * as i1 from '@angular/material/snack-bar';
import { MAT_SNACK_BAR_DATA } from '@angular/material/snack-bar';
import { IconRegistryModule } from '@jeppesen-foreflight/dp-dls-global-angular/icon-registry';

/**
 * Snackbar is a service for displaying snack-bar notifications.
 */
class SnackbarComponent {
    constructor(data, snackBarRef) {
        this.data = data;
        this.snackBarRef = snackBarRef;
        /**
         * The text message to display in the snack bar
         *
         */
        this.message = '';
        /**
         * Snackbar button text
         */
        this.buttonText = '';
        /**
         * A boolean flag determining whether or not to display the close feature
         */
        this.showClose = true;
    }
    onAction() {
        this.snackBarRef.dismissWithAction();
    }
    onClose() {
        this.snackBarRef.dismiss();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "20.3.6", ngImport: i0, type: SnackbarComponent, deps: [{ token: MAT_SNACK_BAR_DATA }, { token: i1.MatSnackBarRef }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "17.0.0", version: "20.3.6", type: SnackbarComponent, isStandalone: true, selector: "ba-snackbar", inputs: { message: "message", action: "action", duration: "duration", buttonText: "buttonText", showClose: "showClose" }, ngImport: i0, template: `
    <div class="ba-snackbar__content">
      <span class="ba-snackbar__message">{{ data.message }}</span>
      <div class="ba-snackbar__actions">
        @if (data.action) {
          <button
            color="accent"
            mat-button
            (click)="onAction()"
            >
            {{ data.action }}
          </button>
        }
        @if (data.showClose) {
          <button mat-icon-button (click)="onClose()">
            <mat-icon svgIcon="icon-close"></mat-icon>
          </button>
        }
      </div>
    </div>
    `, isInline: true, styles: [".ba-snackbar__content{display:flex;justify-content:space-between;align-items:center}.ba-snackbar__message{white-space:pre-line;padding-right:16px}.ba-snackbar__actions{display:flex;align-items:center;flex-shrink:0}\n"], dependencies: [{ kind: "ngmodule", type: MatButtonModule }, { kind: "component", type: i2.MatButton, selector: "    button[matButton], a[matButton], button[mat-button], button[mat-raised-button],    button[mat-flat-button], button[mat-stroked-button], a[mat-button], a[mat-raised-button],    a[mat-flat-button], a[mat-stroked-button]  ", inputs: ["matButton"], exportAs: ["matButton", "matAnchor"] }, { kind: "component", type: i2.MatIconButton, selector: "button[mat-icon-button], a[mat-icon-button], button[matIconButton], a[matIconButton]", exportAs: ["matButton", "matAnchor"] }, { kind: "ngmodule", type: MatIconModule }, { kind: "component", type: i3.MatIcon, selector: "mat-icon", inputs: ["color", "inline", "svgIcon", "fontSet", "fontIcon"], exportAs: ["matIcon"] }, { kind: "ngmodule", type: IconRegistryModule }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "20.3.6", ngImport: i0, type: SnackbarComponent, decorators: [{
            type: Component,
            args: [{ selector: 'ba-snackbar', imports: [MatButtonModule, MatIconModule, IconRegistryModule], template: `
    <div class="ba-snackbar__content">
      <span class="ba-snackbar__message">{{ data.message }}</span>
      <div class="ba-snackbar__actions">
        @if (data.action) {
          <button
            color="accent"
            mat-button
            (click)="onAction()"
            >
            {{ data.action }}
          </button>
        }
        @if (data.showClose) {
          <button mat-icon-button (click)="onClose()">
            <mat-icon svgIcon="icon-close"></mat-icon>
          </button>
        }
      </div>
    </div>
    `, styles: [".ba-snackbar__content{display:flex;justify-content:space-between;align-items:center}.ba-snackbar__message{white-space:pre-line;padding-right:16px}.ba-snackbar__actions{display:flex;align-items:center;flex-shrink:0}\n"] }]
        }], ctorParameters: () => [{ type: undefined, decorators: [{
                    type: Inject,
                    args: [MAT_SNACK_BAR_DATA]
                }] }, { type: i1.MatSnackBarRef }], propDecorators: { message: [{
                type: Input
            }], action: [{
                type: Input
            }], duration: [{
                type: Input
            }], buttonText: [{
                type: Input
            }], showClose: [{
                type: Input
            }] } });

/**
 * Generated bundle index. Do not edit.
 */

export { SnackbarComponent };
